lib/task_helper.rb

```diff
@@ -0,0 +1,40 @@
+require 'json'
+
+class TaskHelper
+  class Error < RuntimeError
+    attr_reader :kind, :details, :issue_code
+
+    def initialize(msg, kind, details = nil)
+      super(msg)
+      @kind = kind
+      @issue_code = issue_code
+      @details = details || {}
+    end
+
+    def to_h
+      { 'kind' =>  kind,
+        'msg' => message,
+        'details' => details }
+    end
+  end
+
+  def self.task(_params = {})
```

Comment:
I think this will be easier to extend it its an instance method. We may want to have multiple functions that update state in the future.

run can still be a class method that creates an instance and calls task on it